pkg/executor/join/index_lookup_hash_join.go

```diff
@@ -411,14 +411,16 @@ func (*indexHashJoinOuterWorker) pushToChan(ctx context.Context, task *indexHash
 	return false
 }
 
-func (e *IndexNestedLoopHashJoin) newOuterWorker(innerCh chan *indexHashJoinTask) *indexHashJoinOuterWorker {
+func (e *IndexNestedLoopHashJoin) newOuterWorker(innerCh chan *indexHashJoinTask, initBatchSize int) *indexHashJoinOuterWorker {
+	maxBatchSize := e.Ctx().GetSessionVars().IndexJoinBatchSize
+	batchSize := min(initBatchSize, maxBatchSize)
 	ow := &indexHashJoinOuterWorker{
 		outerWorker: outerWorker{
 			OuterCtx:         e.OuterCtx,
 			ctx:              e.Ctx(),
 			executor:         e.Children(0),
-			batchSize:        32,
```

Comment:
So is this a bug-fix that the `tidb_max_chunk_size` does not take effect on `IndexHashJoin`?
Not sure if it's by-design.

Reply:
Yes, I think it is not by-design.